Deref in VLAs with not symbolic sizes - II

While https://github.com/esbmc/esbmc/issues/178 has been fixed, it appears only fixed it one dimension.
E.g. this testcase still fails:
```c
void foo(char **q)
{
    q[5];
}

int main(int argc, char **argv)
{
    int a;
    __ESBMC_assume(a >= 5 && a <= 20);
    char *b[2][a + 1]; // notice that this is now a 2d array, was 1d before
    foo(b[0]);
    return 0;
}
```

with
```
ESBMC version 7.8.1 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing aaaa.c
aaaa.c:3:8: warning: expression result unused [-Wunused-value]
    q[5];
    ~ ~^
Converting
Generating GOTO Program
GOTO program creation time: 0.308s
GOTO program processing time: 0.002s
Starting Bounded Model Checking
terminate called after throwing an instance of 'array_type2t::dyn_sized_array_excp'
```

Backtrace:
<details>

```
#0  0x000000000554036c in pthread_kill ()
#1  0x00000000055139c6 in raise ()
#2  0x000000000076245e in abort ()
#3  0x0000000000406b76 in __gnu_cxx::__verbose_terminate_handler() [clone .cold] ()
#4  0x0000000000ce096c in __cxxabiv1::__terminate(void (*)()) ()
#5  0x0000000000ce09d7 in std::terminate() ()
#6  0x0000000000ce0b39 in __cxa_throw ()
#7  0x0000000000c2ea96 in (anonymous namespace)::type_sizet::size_bits (this=0x7fff3ee4c5b0, type=...) at /home/runner/work/esbmc/esbmc/src/util/type_byte_size.cpp:127
#8  0x0000000000c2eeb3 in type_byte_size_bits (type=..., ns=0x0) at /home/runner/work/esbmc/esbmc/src/util/type_byte_size.cpp:167
#9  0x0000000000c2ef0c in type_byte_size (type=..., ns=0x0) at /home/runner/work/esbmc/esbmc/src/util/type_byte_size.cpp:172
#10 0x0000000000adf57f in dereferencet::bounds_check (this=0x7fff3ee4cd40, expr=..., offset=..., type=..., guard=...) at /home/runner/work/esbmc/esbmc/src/pointer-analysis/dereference.cpp:2209
#11 0x0000000000ad6194 in dereferencet::build_reference_to (this=0x7fff3ee4cd40, what=..., mode=..., deref_expr=..., type=..., guard=..., lexical_offset=..., pointer_guard=...)
    at /home/runner/work/esbmc/esbmc/src/pointer-analysis/dereference.cpp:750
#12 0x0000000000ad4a82 in dereferencet::dereference (this=0x7fff3ee4cd40, orig_src=..., to_type=..., guard=..., mode=..., lexical_offset=...) at /home/runner/work/esbmc/esbmc/src/pointer-analysis/dereference.cpp:497
#13 0x0000000000ad32eb in dereferencet::dereference_expr (this=0x7fff3ee4cd40, expr=..., guard=..., mode=...) at /home/runner/work/esbmc/esbmc/src/pointer-analysis/dereference.cpp:170
#14 0x0000000000a333a0 in goto_symext::dereference (this=0xa3b4090, expr=..., mode=...) at /home/runner/work/esbmc/esbmc/src/goto-symex/symex_dereference.cpp:202
#15 0x0000000000a69450 in goto_symext::symex_other (this=0xa3b4090, code=...) at /home/runner/work/esbmc/esbmc/src/goto-symex/symex_other.cpp:18
#16 0x00000000009e8acb in goto_symext::symex_step (this=0xa3b4090, art=...) at /home/runner/work/esbmc/esbmc/src/goto-symex/symex_main.cpp:377
#17 0x0000000000a7a743 in execution_statet::symex_step (this=0xa3b4090, art=...) at /home/runner/work/esbmc/esbmc/src/goto-symex/execution_state.cpp:295
#18 0x000000000098f656 in reachability_treet::get_next_formula (this=0x9066060) at /home/runner/work/esbmc/esbmc/src/goto-symex/reachability_tree.cpp:544
#19 0x00000000007fcb4d in bmct::run_thread (this=0x7fff3ee4d740, eq=...) at /home/runner/work/esbmc/esbmc/src/esbmc/bmc.cpp:1046
#20 0x00000000007fb341 in bmct::run (this=0x7fff3ee4d740, eq=...) at /home/runner/work/esbmc/esbmc/src/esbmc/bmc.cpp:863
#21 0x00000000007fae94 in bmct::start_bmc (this=0x7fff3ee4d740) at /home/runner/work/esbmc/esbmc/src/esbmc/bmc.cpp:840
#22 0x00000000007bf41e in esbmc_parseoptionst::do_bmc (this=0x7fff3ee4d840, bmc=...) at /home/runner/work/esbmc/esbmc/src/esbmc/esbmc_parseoptions.cpp:1537
#23 0x00000000007b9de7 in esbmc_parseoptionst::doit (this=0x7fff3ee4d840) at /home/runner/work/esbmc/esbmc/src/esbmc/esbmc_parseoptions.cpp:628
#24 0x0000000000b483bb in parseoptions_baset::main (this=0x7fff3ee4d840) at /home/runner/work/esbmc/esbmc/src/util/parseoptions.cpp:71
#25 0x00000000007b0c89 in main (argc=3, argv=0x7fff3ee4dcf8) at /home/runner/work/esbmc/esbmc/src/esbmc/main.cpp:11
```
</details>

--- a/regression/esbmc/github_2360/main.c
+++ b/regression/esbmc/github_2360/main.c
@@ -1,0 +1,13 @@
+void foo(char **q)
+{
+    q[5];
+}
+
+int main(int argc, char **argv)
+{
+    int a;
+    __ESBMC_assume(a >= 5 && a <= 20);
+    char *b[2][a + 1]; // notice that this is now a 2d array, was 1d before
+    foo(b[0]);
+    return 0;
+}

--- a/regression/esbmc/github_2360/test.desc
+++ b/regression/esbmc/github_2360/test.desc
@@ -1,0 +1,4 @@
+CORE
+main.c
+--incremental-bmc
+^VERIFICATION SUCCESSFUL$

